src/features/web-compat.js

```diff
@@ -92,6 +92,64 @@ function safariObjectFix () {
     }
 }
 
+/**
+ * Notification fix for adding missing API for Android WebView.
+ */
+function notificationFix () {
+    const Notification = () => {
+        // noop
+    }
+    Notification.requestPermission = () => {
+        return Promise.resolve({ permission: 'denied' })
+    }
+    Notification.permission = 'denied'
+    Notification.maxActions = 2
+
+    // Expose the API
+    // @ts-expect-error window.Notification isn't assignable
+    window.Notification = Notification
+}
+
+/**
+ * Adds missing permissions API for Android WebView.
+ */
+function permissionsFix (settings) {
+    if (window.navigator.permissions) {
+        return
+    }
+    // @ts-expect-error window.navigator isn't assignable
+    window.navigator.permissions = {}
+    class PermissionStatus extends EventTarget {
```

Comment:
Extending `EventTarget` gives us `addEventListener` etc. TS catches this as well if it doesn't exist.